### PR TITLE
Qingwei/improve extremes

### PR DESCRIPTION
**Pull request by qingweibinary**
Now setExtremes will always show barrier, although it is not obvious, I think we should plot a line to indicate barrier with value altogether, is this WIP?

### CODE REVIEW

**Comment by borisyankov** [comment]
I merged it, but will probably need to rework stuff.
Barrier indicator is WIP, yes.

File: src/config/updateExtremes.js [hunk 1]
```diff
@@ -62,9 +64,36 @@ const updateExtremesYAxis = (axis, contract, ticks) => {
...
-    updateExtremesXAxis(chart.xAxis[0], contract);
-    updateExtremesYAxis(chart.yAxis[0], contract, ticks);
+    const xAxis = chart.xAxis[0];
+    const xMin = xAxis.min;
+    const xMax = xAxis.max;
+
+    const zoomedTicks = ticks
+        .filter(t => (t.epoch * 1000) > xMin && (t.epoch * 1000) < xMax);
+
+    let ticksMin = 0;
```
Comment (borisyankov) [inline src/config/updateExtremes.js]: Shouldn't all this code be in updateExtremesYAxis?

File: src/config/updateExtremes.js [hunk 2]
```diff
@@ -62,9 +64,36 @@ const updateExtremesYAxis = (axis, contract, ticks) => {
...
+    const boundaries = [ticksMin, ticksMax, low_barrier, high_barrier].filter(x => x || x === 0);
+
+    const nextMin = arrayMin(boundaries);
+    const nextMax = arrayMax(boundaries);
+
+    const yAxis = chart.yAxis[0];
+    yAxis.setExtremes(nextMin, nextMax);
 };
+
+export default throttle(updateExtremes, 500);
```
Comment (borisyankov) [inline src/config/updateExtremes.js]: Previous updateExtremesYAxis checks for old extremes and updates only if needed.
Why is throttling needed?

File: src/parts/reset.js [hunk 3]
```diff
@@ -9,5 +9,10 @@ export default () => ({
         spacingTop: 0,
         spacingLeft: 0,
         spacingRight: 3,
+        events: {
+            load: function() {
+                this.xAxis[0].chart = this
```
Comment (borisyankov) [inline src/parts/reset.js]: Why is that needed?

File: src/config/updateExtremes.js (hunk 1 above)
Comment (qingweibinary) [inline src/config/updateExtremes.js]: If we do not want to control x extreme that another indirect function call would be unnecessary?

File: src/parts/reset.js (hunk 3 above)
Comment (qingweibinary) [inline src/parts/reset.js]: Because we need to access chart object in xAxis

Comment (borisyankov) [inline src/parts/reset.js]: It already has this property I think?

File: src/config/updateExtremes.js (hunk 1 above)
Comment (borisyankov) [inline src/config/updateExtremes.js]: We certainly do want to control it.
When you show a statement and open contract.

File: src/parts/reset.js (hunk 3 above)
Comment (qingweibinary) [inline src/parts/reset.js]: Nope, it does not.

File: src/config/updateExtremes.js (hunk 1 above)
Comment (qingweibinary) [inline src/config/updateExtremes.js]: I'll refactor it
